src/Sparrow/LowMemory/LowMemoryNotification.cs

```diff
@@ -142,85 +142,98 @@ public static Size GetCurrentProcessMemoryMappedShared()
+                                    AddLowMemEvent(LowMemReason.LowMemOnTimeoutChk, availableMem, totalUnmanagedAllocations,
+                                        memInfo.TotalPhysicalMemory.GetValue(SizeUnit.Bytes));
+                                }
+                                LowMemoryState = true;
+                                clearInactiveHandlersCounter = 0;
+                                RunLowMemoryHandlers(true);
+                                timeout = 500;
+                            }
+                            else
+                            {
+                                if (LowMemoryState)
+                                {
+                                    if (_logger.IsInfoEnabled)
+                                        _logger.Info("Back to normal memory usage detected");
+                                    AddLowMemEvent(LowMemReason.BackToNormal, availableMem, totalUnmanagedAllocations, memInfo.TotalPhysicalMemory.GetValue(SizeUnit.Bytes));
+                                }
+                                LowMemoryState = false;
+                                RunLowMemoryHandlers(false);
+                                timeout = availableMem < _lowMemoryThreshold * 2 ? 1000 : 5000;
+                            }
+                            break;
+                        case 0:
+                            _simulatedLowMemory.Reset();
+                            LowMemoryState = !LowMemoryState;
+                            var memInfoForLog = MemoryInformation.GetMemoryInfo();
+                            var availableMemForLog = memInfoForLog.AvailableMemory.GetValue(SizeUnit.Bytes);
+                            AddLowMemEvent(LowMemoryState ? LowMemReason.LowMemStateSimulation : LowMemReason.BackToNormalSimulation, availableMemForLog,
+                                totalUnmanagedAllocations, memInfoForLog.TotalPhysicalMemory.GetValue(SizeUnit.Bytes));
+                            if (_logger.IsInfoEnabled)
+                                _logger.Info("Simulating : " + (LowMemoryState ? "Low memory event" : "Back to normal memory usage"));
+                            RunLowMemoryHandlers(LowMemoryState);
+                            break;
+                        case 2: // check allocations
+                            _warnAllocation.Reset();
+                            goto case WaitHandle.WaitTimeout;
+                        case 1: // shutdown requested
+                            return;
+                        default:
+                            return;
+                    }
                 }
             }
+            catch (Exception e)
+            {
+                if (_logger.IsInfoEnabled)
+                {
+                    _logger.Info($"Catastrophic failure in low memory notification: {e.Message}", e);
+                }
+
+                throw;
```

Comment:
You cannot throw here, this will allow the exception to escape the thread and kill the process

Reply:
This was the previous behavior.
Fixed. The low memory notification will stop working.